### ISSUE TITLE
Support time-series analysis over a subset of the output data

### ISSUE THREAD

**Issue by xylar** [MPAS-Dev/MPAS-Analysis#45]
Currently, the the code will only work as expected if config option `timeseries_yr1 = 1` and `timeseries_yr2` is set to something beyond the end of the current run (e.g. `9999`), which are the default settings.  If `timeseries_yr1 > 1`, the code crashes, likely because `mpas_xarray` is expecting the time series to start from year 1 (it isn't currently making use of `timeseries_yr1`).  If `timeseries_yr1 = 1` and `timeseries_yr2` is set to be less than the total duration of the run, the timeseries analysis may be performed beyond the desired end time.  This is because we add one file with date stamp beyond the final date to be sure we're not missing any important output.

The time axis of the xarray data sets should be sliced to be consistent with these two new config options in each timeseries analysis function.

**Comment by pwolfram** [comment]
@milenaveneziani and @xylar, we will have some other potential `mpas_xarray` changes too.  Is this something we need to fix now or can we wait until it is time to make the other `mpas_xarray` changes?

**Comment by pwolfram** [comment]
To be clear, this would be for the item:

- [ ] mpas_xarray needs to support different approaches to specification of time output for AMs (#33, #38, others?)

**Comment by xylar** [comment]
@pwolfram, it's not clear to me whether the problem here is that mpas_xarray needs to be updated or just that the *calls* to mpas_xarray need to be updated.  I suspect the latter, in which case this fix is independent of other changes to mpas_xarray.

**Comment by xylar** [comment]
So I think most (perhaps all?) of what is needed to address this issue is to change lines like the following:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/develop/mpas_analysis/ocean/ohc_timeseries.py#L90
```python
    time_start = datetime.datetime(yr_offset+1,1,1)
    time_end = datetime.datetime(yr_offset+1,12,31)
    ds_yr1 = ds.sel(Time=slice(time_start,time_end))
```
to look more like the corresponding lines in the climatology computations, e.g.:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/develop/mpas_analysis/ocean/ocean_modelvsobs.py#L129
```python
    time_start = datetime.datetime(yr_offset+climo_yr1, 1, 1)
    time_end = datetime.datetime(yr_offset+climo_yr2, 12, 31)
    ds_tslice = ds.sel(Time=slice(time_start, time_end))
```

**Comment by milenaveneziani** [comment]
This was addressed by #48, right?

**Comment by pwolfram** [comment]
Please reopen if this is a premature closing @xylar.
